apiserver/facades/client/cloud/cloudV3_test.go

```diff
@@ -0,0 +1,295 @@
+	s.apiv3 = client
+}
+
+func (s *cloudSuiteV3) TestCredentialContentsAllNoSecrets(c *gc.C) {
+	// Get all credentials with no secrets.
+	results, err := s.apiv3.CredentialContents(params.CredentialContentRequestParam{})
+	c.Assert(err, jc.ErrorIsNil)
+	s.backend.CheckCallNames(c, "AllCloudCredentials",
+		"Cloud", "CredentialModelsAndOwnerAccess",
+		"Cloud", "CredentialModelsAndOwnerAccess",
+		"Cloud", "CredentialModelsAndOwnerAccess")
+
+	expected := []params.CredentialContentResult{
+		params.CredentialContentResult{
+			Result: &params.ControllerCredentialInfo{
+				Content: params.CredentialContent{
+					Name:     "onecredential",
+					Cloud:    "dummy",
+					AuthType: "userpass",
+					Attributes: map[string]string{
+						"username": "admin",
+					},
+				},
+				Models: []params.ModelAccess{
+					{Model: "abcmodel", Access: "admin"},
+					{Model: "xyzmodel", Access: "read"},
+				},
+			},
+		},
+		params.CredentialContentResult{
+			Result: &params.ControllerCredentialInfo{
+				Content: params.CredentialContent{
+					Name:     "twocredential",
+					Cloud:    "aws",
+					AuthType: "access-key",
+					Attributes: map[string]string{
+						"access-key": "BLAHB3445635BLAH",
+					},
+				},
+				Models: []params.ModelAccess{
+					{Model: "whynotmodel"}, // no access
+				},
+			},
+		},
+		params.CredentialContentResult{
+			Result: &params.ControllerCredentialInfo{
+				Content: params.CredentialContent{
+					Name:       "mcredential",
+					Cloud:      "maas",
+					AuthType:   "oauth1",
+					Attributes: map[string]string{},
+				},
+				Models: []params.ModelAccess{},
+			},
+		},
+	}
+
+	c.Assert(results.Results, gc.DeepEquals, expected)
+}
+
+func (s *cloudSuiteV3) TestCredentialContentsNoneForUser(c *gc.C) {
+	s.backend.credentials = nil
+	results, err := s.apiv3.CredentialContents(params.CredentialContentRequestParam{})
+	c.Assert(err, jc.ErrorIsNil)
+	s.backend.CheckCallNames(c, "AllCloudCredentials")
+	c.Assert(results.Results, gc.DeepEquals, []params.CredentialContentResult{})
+}
+
+func (s *cloudSuiteV3) TestCredentialContentsNamedWithSecrets(c *gc.C) {
+	results, err := s.apiv3.CredentialContents(params.CredentialContentRequestParam{
+		IncludeSecrets: true,
+		Credentials:    []params.CloudCredentialRequestParam{{CloudName: "aws", CredentialName: "twocredential"}},
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	s.backend.CheckCallNames(c, "CloudCredential", "Cloud", "CredentialModelsAndOwnerAccess")
+
+	expected := []params.CredentialContentResult{
+		params.CredentialContentResult{
+			Result: &params.ControllerCredentialInfo{
+				Content: params.CredentialContent{
+					Name:     "twocredential",
+					Cloud:    "aws",
+					AuthType: "access-key",
+					Attributes: map[string]string{
+						"access-key": "BLAHB3445635BLAH",
+					},
+					Secrets: map[string]string{
+						"secret-key": "fffajdnjsdnnjd667gvd",
+					},
+				},
+				Models: []params.ModelAccess{
+					{Model: "whynotmodel"}, // no access
```

Comment:
Can we use permission.NoAccess explicitly in these tests

Reply:
permission.NoAccess is just an "". Since we are dealing with string here, to use permission.NoAccess, th code will look as ugly as string(permission.NoAccess)